libpod/kube.go

```diff
@@ -0,0 +1,277 @@
+	}
+	ps := v1.PodSpec{
+		Containers: containers,
+	}
+	p := v1.Pod{
+		TypeMeta:   tm,
+		ObjectMeta: om,
+		Spec:       ps,
+	}
+	return &p, nil
+}
+
+// ContainerToV1Container converts information we know about a libpod container
+// to a V1.Container specification.
+func ContainerToV1Container(c *Container) (v1.Container, error) {
+	container := v1.Container{}
+	kubeSec, err := generateKubeSecurityContext(c)
+	if err != nil {
+		return container, err
+	}
+
+	if len(c.Spec().Linux.Devices) > 0 {
+		// TODO Enable when we can support devices and their names
+		devices, err := generateKubeVolumeDeviceFromLinuxDevice(c.Spec().Linux.Devices)
+		if err != nil {
+			return container, err
+		}
+		container.VolumeDevices = devices
+		return container, errors.Wrapf(ErrKubeNotSupported, "linux devices")
+	}
+
+	if len(c.config.UserVolumes) > 0 {
+		// TODO When we until we can resolve what the volume name should be, this is disabled
+		volumes, err := libpodMountsToKubeVolumeMounts(c)
+		if err != nil {
+			return container, err
+		}
+		container.VolumeMounts = volumes
+		return container, errors.Wrapf(ErrKubeNotSupported, "volume names")
+	}
+
+	containerCommands := c.Command()
+	container.Name = removeUnderscores(c.Name())
+
+	_, image := c.Image()
+	container.Image = image
+	container.Stdin = c.Stdin()
+	container.Command = containerCommands
+	// TODO need to figure out how we handle command vs entry point.  Kune appears to prefer entrypoint.
+	// right now we just take the container's command
+	//container.Args = args
+	container.WorkingDir = c.WorkingDir()
+	container.Ports = ocicniPortMappingToContainerPort(c.PortMappings())
+	// This should not be applicable
+	//container.EnvFromSource =
+	container.Env = libpodEnvVarsToKubeEnvVars(c.config.Spec.Process.Env)
+	// TODO enable resources when we can support naming conventions
+	//container.Resources
+	container.SecurityContext = kubeSec
+	container.Stdin = c.Stdin()
+	container.StdinOnce = false
+	container.TTY = c.config.Spec.Process.Terminal
+
+	return container, nil
+}
+
+// ocicniPortMappingToContainerPort takes an ocicni portmapping and converts
+// it to a v1.ContainerPort format for kube output
+func ocicniPortMappingToContainerPort(portMappings []ocicni.PortMapping) []v1.ContainerPort {
+	var containerPorts []v1.ContainerPort
+	for _, p := range portMappings {
+
+		protocol := v1.ProtocolTCP
+		if strings.ToUpper(p.Protocol) == "UDP" {
+			protocol = v1.ProtocolUDP
+		}
+		cp := v1.ContainerPort{
+			// Name will not be supported
+			HostPort:      p.HostPort,
+			HostIP:        p.HostIP,
+			ContainerPort: p.ContainerPort,
+			Protocol:      protocol,
+		}
+		containerPorts = append(containerPorts, cp)
+	}
+	return containerPorts
+}
+
+// libpodEnvVarsToKubeEnvVars converts a key=value string slice to []v1.EnvVar
+func libpodEnvVarsToKubeEnvVars(envs []string) []v1.EnvVar {
+	var envVars []v1.EnvVar
+	for _, e := range envs {
+		splitE := strings.SplitN(e, "=", 2)
```

Comment:
Check if this actually returns an array of length 2 so we don't segfault below

Reply:
fixed